### DISCUSSION TITLE
AR time steps

### DISCUSSION THREAD

**Discussion by DFHydrogeologist** [ourownstory/neural_prophet#240] [Q&A - get help using NeuralProphet] [answered]
Hi!
I'm trying to use AR (using n_lags) in my own dataset and I'm having this error:

`More than 30 consecutive missing values encountered in column y. 163 NA remain. Please preprocess data manually.`

I'm not sure about the difference between one register and another to avoid this error. I have to have data for every calendar day?

**Comment by ourownstory** [comment, marked as answer]
Hi @DFHydrogeologist , 
Yes, if you use AR, you need to have complete data for all timesteps, as they will all become inputs to the model during the training procedure. NP imputes small amounts of missing data automatically. If you have large gaps, it fails in order to protect you from accidentally training on mostly imputed data. Please use an adequate fill-in method to preprocess the data. 
In the future, we might add some support for missing data, but for now you could also train a model per large consecutive block of data that you have.
